source/ingress.go

```diff
@@ -133,6 +145,18 @@ func endpointsFromIngress(ing *v1beta1.Ingress) []*endpoint.Endpoint {
 		if rule.Host == "" {
 			continue
 		}
+
+		// Get the desired hostname of the ingress from the annotation.
+		targetAnnotation, exists := ing.Annotations[targetAnnotationKey]
+		if exists {
```

Comment:
this code block is the same as above and should be extracted into a function to stay DRY